File: lint-rules-rxjava2-lint/src/main/java/com/vanniktech/lintrules/rxjava2/IssueRegistry.java
package com.vanniktech.lintrules.rxjava2;

import com.android.tools.lint.detector.api.Issue;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import static com.vanniktech.lintrules.rxjava2.RxJava2MethodCheckReturnValueDetector.ISSUE_METHOD_MISSING_CHECK_RETURN_VALUE;

public final class IssueRegistry extends com.android.tools.lint.client.api.IssueRegistry {
  @Override public List<Issue> getIssues() {
    final List<Issue> issues = new ArrayList<>();
    Collections.addAll(issues, RxJava2Detector.getIssues());
    Collections.addAll(issues, ISSUE_METHOD_MISSING_CHECK_RETURN_VALUE);
    return issues;
  }
}

File: lint-rules-rxjava2-lint/src/main/kotlin/com/vanniktech/lintrules/rxjava2/IssueRegistry.kt
@file:Suppress("UnstableApiUsage") // We know that Lint APIs aren't final.

package com.vanniktech.lintrules.rxjava2

import com.android.tools.lint.client.api.Vendor
import com.android.tools.lint.detector.api.CURRENT_API

internal const val PRIORITY = 10 // Does not matter anyway within Lint.

class IssueRegistry : com.android.tools.lint.client.api.IssueRegistry() {
  override val api get() = CURRENT_API

  override val vendor get() = Vendor(
    vendorName = "vanniktech/lint-rules/",
    feedbackUrl = "https://github.com/vanniktech/lint-rules/issues",
  )

  override val issues get() = listOf(
    ISSUE_METHOD_MISSING_CHECK_RETURN_VALUE,
    ISSUE_MISSING_COMPOSITE_DISPOSABLE_CLEAR,
    ISSUE_DISPOSABLE_ADD_ALL_CALL,
    ISSUE_DEFAULT_SCHEDULER,
    ISSUE_DISPOSABLE_DISPOSE_CALL,
    ISSUE_SUBSCRIBE_MISSING_ON_ERROR,
    ISSUE_RAW_SCHEDULER_CALL,
  )
}
